src/game/moveFinding/findPawnBaseMoves.js

```diff
@@ -0,0 +1,61 @@
+import BoardCoordinates from "../board/boardCoordinates";
+
+export default function findPawnBaseMoves(boardSituation, startingCoordinates, color) {
+    let actualBoard = boardSituation.position;
+    return findMoves(currentBoard, startingCoordinates, color);
+}
+
+//definition of pawn moves
+
+function findMoves(actualBoard, startingCoordinates, color) {
+    let possibleMoves = []; //Output array
+    let actual = new BoardCoordinates(startingCoordinates.x, startingCoordinates.y);
+    const forwardDirection;
+        color === 'white' ? forwardDirection = 1 : forwardDirection = -1;
+
+    //checking move 2 fields forward
+    const oneForwardTargetCoords = new BoardCoordinates(startingCoordinates.x, startingCoordinates.y + forwardDirection);
+    const twoForwardTargetCoords = new BoardCoordinates(startingCoordinates.x, startingCoordinates.y + 2*forwardDirection);
+    const pawnInStartingPosition = function(twoForwardTargetCoords, color) {
+        if ((color === 'white' && startingCoordinates.y === 1) || (color === 'black' && startingCoordinates.y === 6)) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+    if (pawnInStartingPosition === true && oneForwardTargetCoords == null && twoForwardTargetCoords == null ) {
+        possibleMoves.push(twoForwardTargetCoords);
+    }
+    else {
+        console.log('illegal move');
+    }
+
+    //checking move 1 field forward
+    if (oneForwardTargetCoords == null) {
+        possibleMoves.push(oneForwardTargetCoords);
+    }
+    else {
+        console.log('illegal move');
+    }
+
+    //checking beating diagonal left
+    const diagonalLeftTargetCoords = new BoardCoordinates(startingCoordinates.x - forwardDirection, startingCoordinates.y + forwardDirection);
```

Comment:
Czy nie byłoby lepiej zrobić po prostu +1 oraz -1 (przy ruchu na ukos w drugą stronę)? Kolor pionka nie ma wpływu na to, który ukos sprawdzasz (i tak chcesz sprawdzić oba), kolejność sprawdzania też nie ma znaczenia. Używanie forwardDirection w tej sytuacji jest dziwne - czytelnik kodu może stracić parę minut na zastanawianie się, o co tu chodzi, co to forwardDirection tu robi i jaki ma wpływ na wynik.